example-crates/wallet_esplora_async/src/main.rs

```diff
@@ -24,62 +32,93 @@ async fn main() -> Result<(), anyhow::Error> {
         external_descriptor,
         Some(internal_descriptor),
         db,
-        Network::Testnet,
+        Network::Signet,
     )?;
 
     let address = wallet.try_get_address(AddressIndex::New)?;
-    println!("Generated Address: {}", address);
+    info!("Generated Address: {}", address);
 
     let balance = wallet.get_balance();
-    println!("Wallet balance before syncing: {} sats", balance.total());
-
-    print!("Syncing...");
-    let client =
-        esplora_client::Builder::new("https://blockstream.info/testnet/api").build_async()?;
-
-    let prev_tip = wallet.latest_checkpoint();
-    let keychain_spks = wallet
-        .all_unbounded_spk_iters()
-        .into_iter()
-        .map(|(k, k_spks)| {
-            let mut once = Some(());
-            let mut stdout = std::io::stdout();
-            let k_spks = k_spks
-                .inspect(move |(spk_i, _)| match once.take() {
-                    Some(_) => print!("\nScanning keychain [{:?}]", k),
-                    None => print!(" {:<3}", spk_i),
-                })
```

Comment:
I think the main thing this PR is missing is a way of restoring these prints. I don't think we can lose the ability to monitor progress here. This is the main technical challenge of this improvement. Let me know if you want to discuss. I'm not even sure what I'd do here but I have some ideas.

Reply:
This will be tricky to do so I would like to discuss and get some advice.

Reply:
Can we push the feature to report sync/full_scan progress to a follow-up PR? I think this one is going to have enough in it to review.

Reply:
I'd prefer not to lose features in PRs -- if we don't know how to restore this functionality then that's a deal breaker for this approach so it's better to sort it out now.

Reply:
I wasn't able to print the keychain because there isn't a way to have a generic K in the inspect function and capture it across threads. But I am able to use the index and spk in the function which I did for this example.